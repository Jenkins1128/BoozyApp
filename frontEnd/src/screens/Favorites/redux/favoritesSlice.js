import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
