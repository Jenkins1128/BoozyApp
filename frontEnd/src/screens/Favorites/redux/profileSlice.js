import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
