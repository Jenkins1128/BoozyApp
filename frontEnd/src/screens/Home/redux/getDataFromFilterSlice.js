import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const getDataFromFilterAsync = createAsyncThunk('getDataFromFilterAsync/status', async (data, { rejectWithValue }) => {
	const { location, priceType, cuisine } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${location}/${priceType}/${cuisine}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getDataFromFilterSlice = createSlice({
	name: 'getDataFromFilterAsync',
	initialState,
	reducers: {
		resetDataFromFilterRequestStatus: (state) => {
			state.currentState.dataFromFilterRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getDataFromFilterAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				dataFromFilterRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getDataFromFilterAsync.rejected, (state) => {
			state.currentState.dataFromFilterRequestStatus = 'rejected';
		});
	}
});

export const { resetDataFromFilterRequestStatus } = getDataFromFilterSlice.actions;
export const selectDataFromFilterState = (state) => state.dataFromFilter.currentState;
export default getDataFromFilterSlice.reducer;
