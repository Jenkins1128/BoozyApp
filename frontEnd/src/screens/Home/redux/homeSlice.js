import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {
		location: '',
		cuisine: null,
		priceType: 6,
		$color: 'white',
		$$color: 'white',
		$$$color: 'white',
		$$$$color: 'white',
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: [],
		initialPosition: {
			latitude: 0,
			longitude: 0
		},
		markerPosition: {
			latitude: 0,
			longitude: 0
		}
	}
};

export const getCurrentLocationDataAsync = createAsyncThunk('getCurrentLocationDataAsync/status', async (data, { rejectWithValue }) => {
	const { latitude, longitude } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${latitude}/${longitude}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const homeSlice = createSlice({
	name: 'getCurrentLocationDataAsync',
	initialState,
	reducers: {
		updateLocation: (state, { payload }) => {
			state.currentState.location = payload.location;
		},
		updateCuisine: (state, { payload }) => {
			state.currentState.cuisine = payload.cuisine;
		},
		updatePriceType: (state, { payload }) => {
			state.currentState.priceType = payload.priceType;
		},
		update$color: (state, { payload }) => {
			state.currentState.$color = payload.$color;
		},
		update$$color: (state, { payload }) => {
			state.currentState.$$color = payload.$$color;
		},
		update$$$color: (state, { payload }) => {
			state.currentState.$$$color = payload.$$$color;
		},
		update$$$$color: (state, { payload }) => {
			state.currentState.$$$$color = payload.$$$$color;
		},
		updateAllCategories: (state, { payload }) => {
			state.currentState.allCategories = payload.allCategories;
		},
		updateAlreadyFavorited: (state, { payload }) => {
			state.currentState.alreadyFavorited = payload.alreadyFavorited;
		},
		updateAlreadyRated: (state, { payload }) => {
			state.currentState.alreadyRated = payload.alreadyRated;
		},
		updateRestaurantsArray: (state, { payload }) => {
			state.currentState.restaurantsArray = payload.restaurantsArray;
		},
		updateInitialPosition: (state, { payload }) => {
			state.currentState.initialPosition = payload.initialPosition;
		},
		updateMarkerPosition: (state, { payload }) => {
			state.currentState.markerPosition = payload.markerPosition;
		},
		reset: (state) => {
			Object.assign(state.currentState, {
				location: '',
				cuisine: null,
				priceType: 6,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice1: (state) => {
			Object.assign(state.currentState, {
				priceType: 1,
				$color: 'yellow',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice2: (state) => {
			Object.assign(state.currentState, {
				priceType: 2,
				$color: 'white',
				$$color: 'yellow',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice3: (state) => {
			Object.assign(state.currentState, {
				priceType: 3,
				$color: 'white',
				$$color: 'white',
				$$$color: 'yellow',
				$$$$color: 'white'
			});
		},
		setPrice4: (state) => {
			Object.assign(state.currentState, {
				priceType: 4,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'yellow'
			});
		},
		resetHomeRequestStatus: (state) => {
			state.currentState.locationDataRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getCurrentLocationDataAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					locationDataRequestStatus: 'fulfilled',
					restaurantsArray: payload
				});
			})
			.addCase(getCurrentLocationDataAsync.rejected, (state) => {
				state.currentState.locationDataRequestStatus = 'rejected';
			});
	}
});

export const {
	updateLocation,
	updateCuisine,
	updatePriceType,
	update$color,
	update$$color,
	update$$$color,
	update$$$$color,
	updateAllCategories,
	updateAlreadyFavorited,
	updateAlreadyRated,
	updateRestaurantsArray,
	updateInitialPosition,
	updateMarkerPosition,
	reset,
	setPrice1,
	setPrice2,
	setPrice3,
	setPrice4,
	resetHomeRequestStatus
} = homeSlice.actions;

export const selectHomeState = (state) => state.home.currentState;
export default homeSlice.reducer;
