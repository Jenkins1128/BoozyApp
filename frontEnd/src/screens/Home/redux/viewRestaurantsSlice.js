import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const viewRestaurantsAsync = createAsyncThunk('viewRestaurantsAsync/status', async (data, { rejectWithValue }) => {
	const { index, state } = data;
	state.index = index;
	const restaurantId = state.restaurantsArray[index]['id'];
	const name = state.restaurantsArray[index]['name'];
	try {
		const alreadyFavoritedResponse = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/favorites', {
			restaurantId: restaurantId,
			name: name
		});
		state.alreadyFavorited = alreadyFavoritedResponse.data[0]['contains'];
		const alreadyRatedResponse = await axios.get(`https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/${state.restaurantsArray[index]['id']}/review`);
		state.alreadyRated = alreadyRatedResponse.data[0]['contains'];
		state.allCategories = '';
		for (let i = 0; i < state.restaurantsArray[index]['categories'].length; i++) {
			if (i == 0) {
				state.allCategories += state.restaurantsArray[index]['categories'][i]['title'];
			} else {
				state.allCategories += ', ' + state.restaurantsArray[index]['categories'][i]['title'];
			}
		}
		const newState = { ...state, restaurantsArray: [state.restaurantsArray[index]] };
		return newState;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const viewRestaurantsSlice = createSlice({
	name: 'viewRestaurantsAsync',
	initialState,
	reducers: {
		resetRestaurantRequestStatus: (state) => {
			state.currentState.restaurantRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(viewRestaurantsAsync.fulfilled, (state, { payload }) => {
				state.currentState = { ...state.currentState, restaurantInfo: payload, restaurantRequestStatus: 'fulfilled' };
			})
			.addCase(viewRestaurantsAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, restaurantRequestStatus: 'rejected' };
			});
	}
});

export const { resetRestaurantRequestStatus } = viewRestaurantsSlice.actions;
export const selectRestaurantState = (state) => state.restaurant.currentState;

export default viewRestaurantsSlice.reducer;
