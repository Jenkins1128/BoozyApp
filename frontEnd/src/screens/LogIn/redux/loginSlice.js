import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const loginAsync = createAsyncThunk('loginAsync/status', async (data, { rejectWithValue }) => {
	const { email, password } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/login', {
			email,
			password
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const loginSlice = createSlice({
	name: 'loginAsync',
	initialState,
	reducers: {
		updateEmail: (state, { payload }) => {
			state.currentState.email = payload.email;
		},
		updatePassword: (state, { payload }) => {
			state.currentState.password = payload.password;
		},
		resetStatus: (state) => {
			state.currentState.requestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(loginAsync.pending, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'pending',
					isLoading: true
				});
			})
			.addCase(loginAsync.fulfilled, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'fulfilled',
					isLoading: false
				});
			})
			.addCase(loginAsync.rejected, (state) => {
				Object.assign(state.currentState, {
					requestStatus: 'rejected',
					isLoading: false
				});
			});
	}
});

export const { updateEmail, updatePassword, resetStatus } = loginSlice.actions;
export const selectLogInState = (state) => state.login.currentState;

export default loginSlice.reducer;
