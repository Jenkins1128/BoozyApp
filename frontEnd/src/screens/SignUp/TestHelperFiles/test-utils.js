import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
