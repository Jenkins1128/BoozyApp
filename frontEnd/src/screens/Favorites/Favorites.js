import React, { useEffect } from 'react';
import { StyleSheet, Text, FlatList, View, Alert, SafeAreaView, StatusBar } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { getOS } from '../../helpers/os';
import Favorite from './Favorite/Favorite';
import { getFavoritesAsync, resetGetFavoritesRequestStatus, selectFavoritesState } from './redux/favoritesSlice';

const Favorites = ({ navigation }) => {
	const state = useSelector(selectFavoritesState);
	const dispatch = useDispatch();

	//did screen focus, get restaurant data
	useEffect(() => {
		const unsubscribe = navigation.addListener('focus', () => {
			dispatch(getFavoritesAsync());
		});
		return unsubscribe;
	}, [navigation]);

	useEffect(() => {
		switch (state.getFavoritesRequestStatus) {
			case 'idle':
				return;
			case 'rejected':
				showErrorAlert('Unable to get favorites. Please check your network.');
				break;
			case 'fulfilled':
				break;
			default:
				return;
		}
		dispatch(resetGetFavoritesRequestStatus());
	}, [state]);

	const showErrorAlert = (errorString) => {
		Alert.alert('Uh oh', errorString, [{ text: 'OK' }]);
	};

	const favorite = ({ item, index }) => {
		return <Favorite key={index} keyval={index} val={item} />;
	};

	return (
		<SafeAreaView testID={'container'} style={styles.container}>
			<View testID={'myFavoritesContainer'} style={styles.myFavoritesContainer}>
				<Text testID={'myFavoritesText'} style={styles.myFavoritesText}>
					My Favorites
				</Text>
			</View>
			<View testID={'myFavoritesListContainer'} style={styles.myFavoritesListContainer}>
				<FlatList data={state.restaurantsArray} keyExtractor={(item, i) => i.toString()} renderItem={favorite} />
			</View>
		</SafeAreaView>
	);
};

export default Favorites;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingTop: getOS() === 'android' ? StatusBar.currentHeight : 0,
		backgroundColor: '#fff',
		alignItems: 'center',
		justifyContent: 'center',
		width: '100%'
	},
	myFavoritesContainer: {
		flex: 1
	},
	myFavoritesListContainer: {
		flex: 7
	},
	myFavoritesText: {
		color: '#EB8873',
		fontFamily: 'Arial',
		fontWeight: 'bold',
		fontSize: 30
	}
});
