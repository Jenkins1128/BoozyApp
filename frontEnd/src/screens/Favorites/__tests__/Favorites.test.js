import React from 'react';
import { reduxRender, cleanup } from '../../../TestHelperFiles/redux/test-utils';
import Favorites from '../Favorites';
import FavoritesTestFuncs from '../../../TestHelperFiles/Functions/FavoritesTestFuncs';
import axios from 'axios';
import { configureStore } from '@reduxjs/toolkit';
import { baseUrl } from '../../../helpers/constants';
import favoritesReducer, { getFavoritesAsync } from '../redux/favoritesSlice';

const createTestProps = (props) => ({
	navigation: {
		navigate: jest.fn(),
		addListener: jest.fn()
	},
	...props
});

describe('<Favorites />', () => {
	let props;
	let rendered;

	beforeEach(() => {
		props = createTestProps({});
		rendered = reduxRender(<Favorites {...props} />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should wrap view with a flexible wrapper', () => {
		const containerComponent = rendered.getByTestId('container');
		const given = containerComponent.props.style;
		const result = {
			flex: 1,
			paddingTop: 0,
			backgroundColor: '#fff',
			alignItems: 'center',
			justifyContent: 'center',
			width: '100%'
		};
		expect(result).toMatchObject(given);
	});

	it('should fill parent container 1/8', () => {
		const myFavoritesContainerComponent = rendered.getByTestId('myFavoritesContainer');
		const given = myFavoritesContainerComponent.props.style;
		const result = {
			flex: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should fill parent container 7/8', () => {
		const myFavoritesContainerComponent = rendered.getByTestId('myFavoritesListContainer');
		const given = myFavoritesContainerComponent.props.style;
		const result = {
			flex: 7
		};
		expect(result).toMatchObject(given);
	});

	it('should render myFavoritesText styles', () => {
		const myFavoritesTextComponent = rendered.getByTestId('myFavoritesText');
		const given = myFavoritesTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontFamily: 'Arial',
			fontWeight: 'bold',
			fontSize: 30
		};
		expect(result).toMatchObject(given);
	});

	//functions
	describe('showErrorAlert', () => {
		it('calls sideEffect', () => {
			jest.spyOn(FavoritesTestFuncs, 'showErrorAlert');
			FavoritesTestFuncs.showErrorAlert('Network failed...');
			expect(FavoritesTestFuncs.showErrorAlert).toHaveBeenCalledTimes(1);
			expect(FavoritesTestFuncs.showErrorAlert).toHaveBeenCalledWith('Network failed...');
		});
	});

	describe('favorite', () => {
		it('returns Favorite component', () => {
			const restaurant = {
				index: 0,
				item: {}
			};
			jest.spyOn(FavoritesTestFuncs, 'favorite');
			FavoritesTestFuncs.favorite(restaurant);
			expect(FavoritesTestFuncs.favorite).toHaveReturned();
			expect(FavoritesTestFuncs.favorite).toHaveBeenCalledWith(restaurant);
		});
	});

	//redux integration
	describe('redux', () => {
		it('should dispatch getFavoritesAsync and state updated fulfilled', async () => {
			const getSpy = jest.spyOn(axios, 'get').mockResolvedValueOnce([]);
			const store = configureStore({
				reducer: {
					favorites: favoritesReducer
				}
			});
			await store.dispatch(getFavoritesAsync());
			expect(getSpy).toBeCalledWith(`${baseUrl}/get`);
			const state = store.getState();
			expect(state).toEqual({
				favorites: {
					currentState: {
						allCategories: '',
						alreadyFavorited: false,
						alreadyRated: false,
						restaurantsArray: undefined,
						getFavoritesRequestStatus: 'fulfilled'
					}
				}
			});
		});
	});
});
