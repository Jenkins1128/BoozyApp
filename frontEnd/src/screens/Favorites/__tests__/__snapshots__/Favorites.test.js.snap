// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites /> should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 0,
      "width": "100%",
    }
  }
  testID="container"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
    testID="myFavoritesContainer"
  >
    <Text
      style={
        Object {
          "color": "#EB8873",
          "fontFamily": "Arial",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
      testID="myFavoritesText"
    >
      My Favorites
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 7,
      }
    }
    testID="myFavoritesListContainer"
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
